### PR TITLE
Addon for Ansible Service Broker

### PR DESCRIPTION
**Pull request by sabre1041**
Addon for deploying the [Ansible Service Broker](https://github.com/openshift/ansible-service-broker)

### CODE REVIEW

File: add-ons/ansible-service-broker/README.md [hunk 1]
```diff
@@ -0,0 +1,70 @@
+Ansible Service Broker Addon
+======================
```
Comment (robin-owen) [inline add-ons/ansible-service-broker/README.md]: This is a suggestion which you can feel free to ignore, but I find that it's easier to both maintain and understand headings which use `#`.

File: add-ons/ansible-service-broker/README.md [hunk 2]
```diff
@@ -0,0 +1,70 @@
...
+
+Minishift/CDK must be deployed with the Service Catalog enabled
+
+The Service Catalog can be deployed by adding the `--service-catalog` when starting Minishift. The ability to pass extra flags during startup can be enabled by setting the _MINISHIFT_ENABLE_EXPERIMENTAL_ environmental variable as follows:
+
+```
+$ export MINISHIFT_ENABLE_EXPERIMENTAL=y
+```
+
+Start Minishift with the `--service-catalog` flag
```
Comment (robin-owen) [inline add-ons/ansible-service-broker/README.md]: Please add a colon to the end of this sentence.

File: add-ons/ansible-service-broker/README.md [hunk 3]
```diff
@@ -0,0 +1,70 @@
...
+|----|-----------|-------------|
+|`DOCKERHUB_USER`|Username to authenticate to DockerHub|`changeme`|
+|`DOCKERHUB_PASS`|Password to authenticate to DockerHub|`changeme`|
+|`DOCKERHUB_ORG`|Organization to query for Ansible Playbook Bundles in DockerHub|`ansibleplaybookbundle`|
+|`BROKER_KIND`|Kubernetes API Kind|`Broker`|
+|`SVC_CAT_API_VER`|Kubernetes API Version|`servicecatalog.k8s.io/v1alpha1`|
+|`BROKER_AUTH`|Broker Authentication|`{"basicAuthSecret":{"namespace":"ansible-service-broker","name":"asb-auth-secret"}}`|
+|`ENABLE_BASIC_AUTH`|Enable Basic Authentication within the Broker|`true`|
+
+Variables can be specified by adding `--addon-env <key=value>` when the addon is being invoked (`minishift start` or `minishift addons apply`)
```
Comment (robin-owen) [inline add-ons/ansible-service-broker/README.md]:
```adoc
(either by `minishift start` or `minishift addons apply`).
```

Also, please note the period ending the sentence. :)

File: add-ons/ansible-service-broker/README.md [hunk 4]
```diff
@@ -0,0 +1,70 @@
+Ansible Service Broker Addon
+======================
+
+This addon installs the [Ansible Service Broker](https://github.com/openshift/ansible-service-broker)
```
Comment (robin-owen) [inline add-ons/ansible-service-broker/README.md]: We refer to add-ons with hyphenation within the documentation -- please do so throughout this document as well. :)

File: add-ons/ansible-service-broker/README.md [hunk 5]
```diff
@@ -0,0 +1,70 @@
...
+
+# Prerequisites
+
+Minishift/CDK must be deployed with the Service Catalog enabled
+
+The Service Catalog can be deployed by adding the `--service-catalog` when starting Minishift. The ability to pass extra flags during startup can be enabled by setting the _MINISHIFT_ENABLE_EXPERIMENTAL_ environmental variable as follows:
```
Comment (robin-owen) [inline add-ons/ansible-service-broker/README.md]:
```adoc
the `--service-catalog` flag
```

File: add-ons/ansible-service-broker/README.md [hunk 6]
```diff
@@ -0,0 +1,70 @@
...
+|`BROKER_AUTH`|Broker Authentication|`{"basicAuthSecret":{"namespace":"ansible-service-broker","name":"asb-auth-secret"}}`|
+|`ENABLE_BASIC_AUTH`|Enable Basic Authentication within the Broker|`true`|
+
+Variables can be specified by adding `--addon-env <key=value>` when the addon is being invoked (`minishift start` or `minishift addons apply`)
+
+## Apply the Addon
+
+* To apply the addon to a running instance of Minishift, execute the following command:
+
+        $ minishift addons apply ansible-service-broker
```
Comment (robin-owen) [inline add-ons/ansible-service-broker/README.md]: I'd like to see one more new line here to separate the command example from the following sentence.

File: add-ons/ansible-service-broker/README.md [hunk 7]
```diff
@@ -0,0 +1,70 @@
...
+* To enable the addon each time Minishift starts, execute the following command:
+
+        $ minishift addons enable ansible-service-broker
+
+## Remove the Addon
+
+To remove all of the deployed components, execute the following command
```
Comment (robin-owen) [inline add-ons/ansible-service-broker/README.md]: Please add a colon to the end of this sentence.

File: add-ons/ansible-service-broker/README.md [hunk 8]
```diff
@@ -0,0 +1,70 @@
...
+
+```
+$ minishift addons remove ansible-service-broker
+```
+
+## Uninstall the addon
+
+To uninstall the addon, execute the following command
```
Comment (robin-owen) [inline add-ons/ansible-service-broker/README.md]: Please add a colon to the end of this sentence.

**Comment by sabre1041** [comment]
@robin-owen updated based on feedback

File: add-ons/ansible-service-broker/README.md [hunk 9]
```diff
@@ -0,0 +1,71 @@
...
+
+```
+$ export MINISHIFT_ENABLE_EXPERIMENTAL=y
+```
+
+Start Minishift with the `--service-catalog` flag:
+
+```
+$ minishift start --service-catalog
+```
```
Comment (LalatenduMohanty) [inline add-ons/ansible-service-broker/README.md]: @sabre1041 I think OpenShift v3.7.0 is advisable for this add-on as 3.6.0 had different implementation for `oc cluster up` and it did not work always. WDYT?

Comment (LalatenduMohanty) [reply add-ons/ansible-service-broker/README.md]: @sabre1041 tested on v 3.7.0 and it worked for me. Going ahead and merging this, please send another PR if you want to set minimum version to v3.7.0 https://docs.openshift.org/latest/minishift/using/addons.html#addon-openshift-version-semantics
```
$ minishift start --openshift-version v3.7.0 --service-catalog
-- Starting profile 'minishift'
-- Checking if requested hypervisor 'kvm' is supported on this platform ... OK
-- Checking if KVM driver is installed ... 
   Driver is available at /usr/local/bin/docker-machine-driver-kvm ... 
   Checking driver binary is executable ... OK
-- Checking if Libvirt is installed ... OK
-- Checking if Libvirt default network is present ... OK
-- Checking if Libvirt default network is active ... OK
-- Checking the ISO URL ... OK
-- Starting local OpenShift cluster using 'kvm' hypervisor ...
-- Minishift VM will be configured with ...
   Memory:    2 GB
   vCPUs :    2
   Disk size: 20 GB
-- Starting Minishift VM ...................... OK
-- Checking for IP address ... OK
-- Checking if external host is reachable from the Minishift VM ... 
   Pinging 8.8.8.8 ... OK
-- Checking HTTP connectivity from the VM ... 
   Retrieving http://minishift.io/index.html ... OK
-- Checking if persistent storage volume is mounted ... OK
-- Checking available disk space ... 0% used OK
-- Downloading OpenShift binary 'oc' version 'v3.7.0'
 38.51 MiB / 38.51 MiB [=========================================================================================================================================] 100.00% 0s-- Downloading OpenShift v3.7.0 checksums ... OK
-- OpenShift cluster will be configured with ...
   Version: v3.7.0
-- Checking `oc` support for startup flags ... 
   host-config-dir ... OK
   host-data-dir ... OK
   host-pv-dir ... OK
   host-volumes-dir ... OK
   version ... OK
   routing-suffix ... OK
   service-catalog ... OK
Starting OpenShift using openshift/origin:v3.7.0 ...
Pulling image openshift/origin:v3.7.0
Pulled 1/4 layers, 26% complete
Pulled 1/4 layers, 42% complete
Pulled 1/4 layers, 57% complete
Pulled 1/4 layers, 72% complete
Pulled 2/4 layers, 76% complete
Pulled 3/4 layers, 88% complete
Pulled 3/4 layers, 99% complete
Pulled 4/4 layers, 100% complete
Extracting
Image pull complete
OpenShift server started.

The server is accessible via web console at:
    https://192.168.42.113:8443

You are logged in as:
    User:     developer
    Password: <any value>

To login as administrator:
    oc login -u system:admin

[lmohanty@LalatenduM-laptop minishift-addons]$ minishift addon apply ansible-service-broker
-- Applying addon 'ansible-service-broker':.......
Ansible Service Broker Deployed
```

**Pull request merged**